utils/media_clients/embedding_client.py

```
@@ -0,0 +1,84 @@
+
+logger = logging.getLogger(__name__)
+
+
+class EmbeddingClientStrategy(BaseMediaStrategy):
+    """Strategy for embedding models."""
+
+    def run_eval(self) -> None:
+        """Run evaluations for the model."""
+        logger.info(
+            f"Running evals for model: {self.model_spec.model_name} on device: {self.device.name}"
+        )
+        try:
+            health_status, runner_in_use = self.get_health()
+            if health_status:
+                logger.info("Health check passed.")
+            else:
+                logger.error("Health check failed.")
+                return
+
+            logger.info(f"Runner in use: {runner_in_use}")
+
+        except Exception as e:
+            logger.error(f"Eval execution encountered an error: {e}")
+            return
+
+        logger.info("Generating eval report...")
+        benchmark_data = {}
+
+        # Write benchmark_data to JSON file
+        eval_filename = (
+            Path(self.output_path)
+            / f"eval_{self.model_spec.model_id}"
+            / self.model_spec.hf_model_repo.replace("/", "__")
+            / f"results_{time.time()}.json"
+        )
+        # Create directory structure if it doesn't exist
+        eval_filename.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(eval_filename, "w") as f:
+            json.dump(benchmark_data, f, indent=4)
+        logger.info(f"Evaluation data written to: {eval_filename}")
+
+    def run_benchmark(self, attempt=0) -> list[AudioTestStatus]:
+        """Run benchmarks for the model."""
+        logger.info(
+            f"Running benchmarks for model: {self.model_spec.model_name} on device: {self.device.name}"
+        )
+        try:
+            health_status, runner_in_use = self.get_health()
+            if health_status:
+                logger.info(f"Health check passed. Runner in use: {runner_in_use}")
+            else:
+                logger.error("Health check failed.")
+                return []
```

We should `raise` here instead of silent failure